CMakeLists.txt

```diff
@@ -1,4 +1,6 @@
-cmake_minimum_required(VERSION 3.15)
+cmake_minimum_required(VERSION 3.18)
+
+message(STATUS "The cmake version is ${CMAKE_VERSION}")
```

Comment:
Please remove this line.

Reply:
Printing compiler versions are also not necessary.
The user can do `cmake --version` to find out what version of cmake the program will use to build.
Where does it stop?
Why not print the user environment then.

Reply:
Because printing the entire user environment is not useful.

But printing those variables from the user environment that build depends on, like versions of the compilers, libraries, and cmake version can be useful while debugging failed build. Especially on non tier-1 platforms (i.e. personal laptops and other so called generic platforms) for which we don't know exactly which version are used.